XenModel/Actions/USB/CreateVUSBAction.cs

```diff
@@ -41,19 +41,28 @@ namespace XenAdmin.Actions
     public class CreateVUSBAction : PureAsyncAction
     {
         private PUSB _pusb;
-        private VM _vm;
-        private Dictionary<string, string> _other_config = null;
+        private Dictionary<string, string> _platform;
 
-        public CreateVUSBAction(PUSB pusb, VM vm) : 
+        public CreateVUSBAction(PUSB pusb, VM vm, Dictionary<string, string> platform) : 
             base(pusb.Connection, String.Format(Messages.ACTION_VUSB_CREATING,  pusb.Name(), vm.Name()))
         {
             _pusb = pusb;
-            _vm = vm;
+            VM = vm;
+            _platform = platform == null ?

```

Comment:
We don't need to pass the platform as a parameter to the action, because we have the VM object and can simply check the value of `VM.platform` directly where needed (so we would have `platform` as a local variable in `Run()` and create it in there from `VM.platform`)